components/public-api/gitpod/v1/prebuild.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+import "gitpod/v1/pagination.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service PrebuildService {
+    rpc StartPrebuild(StartPrebuildRequest) returns (StartPrebuildResponse) {}
+    rpc StopPrebuild(StopPrebuildRequest) returns (StopPrebuildResponse) {}
+
+    rpc GetPrebuild(GetPrebuildRequest) returns (GetPrebuildResponse) {}
+    rpc ListPrebuilds(ListPrebuildsRequest) returns (ListPrebuildsResponse) {}
+    rpc WatchPrebuild(WatchPrebuildRequest) returns (stream WatchPrebuildResponse) {}
+}
+
+message GetPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+message GetPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message ListPrebuildsRequest {
+    PaginationRequest pagination = 1;
+    string prebuild_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+
+message ListPrebuildsResponse {
+    PaginationResponse pagination = 1;
+    repeated Prebuild prebuilds = 2;
+}
+
+message StartPrebuildRequest {
+    string configuration_id = 1;
+    string branch = 2;
+}
+message StartPrebuildResponse {
+    string prebuild_id = 1;
+}
+
+message StopPrebuildRequest {
+    string prebuild_id = 1;
+    // waht for?
+    string configuration_id = 2;
+}
+message StopPrebuildResponse {
+}
+
+message WatchPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+}
+message WatchPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message Author {
+    string name = 1;
+    string avatar_url = 2;
+}
+
+message Commit {
```

Comment:
I'm not sure about names of this shapes. In JSON-RCP we called it Change but I could not undertand what it mean, when to implementation and figure out that it simply a commit, so decide to remove indireaction Any objections with it?

I wonder where we should put such global types, like Commit and Author.

Reply:
Judging by the fields this looks more like "CommitMetadata". WDYT?

Reply:
> I wonder where we should put such global types, like Commit and Author.

I'm not sure there will be much use; maybe if we group them under "Git types"...? But I don't see them used anywhere atm. Maybe in ContextService, though.

Reply:
> Judging by the fields this looks more like "CommitMetadata". WDYT?

I called it commit after GitHub API actually: https://docs.github.com/en/graphql/reference/objects#commit. I don't have strong opinion, Commit sounds fine to me as well.